approaches/bulk-data-loading/src/GqlMutations.js

```diff
@@ -0,0 +1,85 @@
+export class GqlMutations {
+
+    static createInventoryListQuery() {
+        let query = `
+        query ListInventoryItems {
+          list_InventoryItems(readMode: NODE_LEDGERED) {
+            _InventoryItems {
+              _id
+            }
+            nextToken
+          }
+        }
+    `;
+
+        return {
+            query: query,
+            variables: {}
+        }
+    }
+
+    static createInventoryListNextPageQuery(nextToken) {
+        let query = `
+        query ListInventoryItems($nextToken: String) {
+          list_InventoryItems(nextToken: $nextToken, readMode: NODE_LEDGERED) {
+            _InventoryItems {
+              _id
+            }
+            nextToken
+          }
+        }
+    `;
+
+        return {
+            query: query,
+            variables: {
+                nextToken: nextToken
+            }
+        }
+    }
+
+    static createInventoryMutation(batch) {
+        let variables = {}
+        let mutationParameters = []
+        let mutationOperations = []
+
+        batch.map((batchRecord, index) => {
+            variables["input" + index] = batchRecord
+            mutationParameters.push(GqlMutations.createMutationParameter(index))
+            mutationOperations.push(GqlMutations.createMutationOperation(index))
+        })
+
+        return {
+            query: GqlMutations.createMutationString(mutationParameters, mutationOperations),
+            variables: variables
+        }
+    }
+
+    static createMutationParameter(index) {
+        return "$input" + index + ": Self_Inventory_Input_!";
+    }
+
+    static createMutationOperation(index) {
```

Comment:
nit: there's a lot of string munging in here. Going forward we should ideally make higher level abstractions in the client SDK

Reply:
Agreed.